wsd/ProxyRequestHandler.cpp

```diff
@@ -0,0 +1,99 @@
+/* -*- Mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4; fill-column: 100 -*- */
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <config.h>
+
+#include <Poco/URI.h>
+
+#include <COOLWSD.hpp>
+#include "ProxyRequestHandler.hpp"
+#include <net/HttpRequest.hpp>
+#include <net/HttpHelper.hpp>
+
+std::map<unsigned long, std::shared_ptr<StreamSocket>> ProxyRequestHandler::LinkSession;
+std::map<std::string, std::shared_ptr<http::Response>> ProxyRequestHandler::CacheFileHash;
+std::chrono::system_clock::time_point ProxyRequestHandler::MaxAge;
+
+void ProxyRequestHandler::handleRequest(const std::string& relPath,
+                                        const std::shared_ptr<StreamSocket>& socket)
+{
+    Poco::URI uriProxy(ProxyServer);
+    const auto timeNow = std::chrono::system_clock::now();
+
+    if (MaxAge > std::chrono::system_clock::time_point() && timeNow > MaxAge)
+    {
+        CacheFileHash.clear();
+        MaxAge = std::chrono::system_clock::time_point();
+    }
+
+    const auto cacheEntry = CacheFileHash.find(relPath);
+    if (cacheEntry != CacheFileHash.end())
+    {
+        socket->sendAndShutdown(*cacheEntry->second.get());
+        return;
+    }
+
+    uriProxy.setPath(relPath);
+    auto sessionProxy = http::Session::create(uriProxy.getHost(),
+                                              http::Session::Protocol::HttpSsl,
+                                              uriProxy.getPort());
+    sessionProxy->setTimeout(std::chrono::seconds(10));
+    http::Request requestProxy(uriProxy.getPathAndQuery());
+    http::Session::FinishedCallback proxyCallback =
+        [](const std::shared_ptr<http::Session>& httpSession)
+            {
+                const auto callbackNow = std::chrono::system_clock::now();
+                auto sessionLink = LinkSession.find(reinterpret_cast<unsigned long>(httpSession.get()));
```

Comment:
We could use `void*` and avoid this cast. But, even better (and good news!) we don't need this map _at all_.

We simply need to capture the `std::shared_ptr<StreamSocket>` in this lambda! Simply change line 47 with this:
```
       [socket](const std::shared_ptr<http::Session>& httpSession)
```
And rename `sessionLink` to `socket` in the code below.

Reply:
Yes, I did before capturing the socket, I wanted to try some alternatives. It seems reasonable to capture the socket
to simplify the code